File: src/test/java/fi/metatavu/metaform/test/functional/tests/GeneralTestProfile.java
package fi.metatavu.metaform.test.functional.tests;

import io.quarkus.test.junit.QuarkusTestProfile;

import java.util.Collections;
import java.util.HashMap;
import java.util.Map;

/**
 * Test profile settings common values for all tests
 */
public class GeneralTestProfile implements QuarkusTestProfile {

  @Override
  public Map<String, String> getConfigOverrides() {
    Map<String, String> properties = new HashMap<String, String>();
    properties.put("runmode", "TEST");
    properties.put("metaform.uploads.folder", "/tmp");
    properties.put("quarkus.liquibase.contexts", "test");
    return properties;
  }
}

File: src/test/java/fi/metatavu/metaform/server/test/functional/tests/GeneralTestProfile.kt
package fi.metatavu.metaform.server.test.functional.tests

import io.quarkus.test.junit.QuarkusTestProfile

/**
 * Test profile settings common values for all tests
 */
class GeneralTestProfile : QuarkusTestProfile {
    override fun getConfigOverrides(): Map<String, String> {
        val properties: MutableMap<String, String> = HashMap()
        properties["runmode"] = "TEST"
        properties["metaform.uploads.folder"] = "/tmp"
        properties["quarkus.liquibase.contexts"] = "test"
        properties["metaforms.keycloak.card.identity.provider"] = "oidc"
        properties["metaforms.features.auditlog"] = "true"
        properties["metaforms.features.cardauth"] = "true"
        properties["metaforms.environment"] = "DEVELOPMENT"
        properties["metaforms.deletion.interval"] = "3s"
        properties["metaforms.deletion.delay"] = "1s"
        return properties
    }
}